lib/spack/spack/build_systems/intel.py

```diff
@@ -1241,6 +1247,33 @@ def configure_rpath(self):
             with open(compiler_cfg, 'w') as fh:
                 fh.write('-Xlinker -rpath={0}\n'.format(compilers_lib_dir))
 
+    # Need a copy of the auto_dispatch_options to be used in the subclasses
+    auto_dispatch_options = auto_dispatch_options
```

Comment:
You don't need this and the global `auto_dispatch_options`. If you use the global the other classes using this can refer to it like `intel.auto_dispatch_options`.

A few notes:

* python variables are usually references so this would be referring to the same thing
* tuples are immutable
* below where you say `for x in auto_dispatch_options` I'm not actually sure whether that will resolve to this or to the global definition. Either way, you don't need both.

Reply:
Right, I will push out a change for this. I think I confused myself with the global variable. 😕